### ISSUE TITLE
Support for the AWS KMS encryption type

### ISSUE THREAD

**Issue by MartinFalatic** [etianen/django-s3-storage#72]
This package currently supports AES256 encryption. AWS KMS should be supported as well.

Currently this package uses the boolean AWS_S3_ENCRYPT_KEY to mediate whether to use no encryption or to use AES256 encryption.

A backward compatible change would add another variable (e.g., AWS_S3_ENCRYPT_KEY_ID) to specify the key type iff AWS_S3_ENCRYPT_KEY is true.

A somewhat breaking change would be to use AWS_S3_ENCRYPT_KEY such that if it's empty or None then encryption is disabled, otherwise the encryption type == the value of AWS_S3_ENCRYPT_KEY. This should still act correctly if encryption isn't being used, but if it is, then the AES256 case would need to be explicitly stated (or assumed to be the case if the value is  a boolean value of True).

Edit: in addition to `ServerSideEncryption`, optional params `Key` and `SSEKMSKeyId` would need to be handled as well.

Your thoughts?

**Comment by etianen** [comment]
Good idea. I like keeping just the one setting, and being backwards
compatible, so that:

   - Anything *falsy* means disabled.
   - A string means "use this encryption method".
   - Anything else truthy means: "use AES256".

I'm happy to take a pull request for this.

**Comment by MartinFalatic** [comment]
I'm working on it. However, when AWS KMS is used, it's unclear how to properly pass the AWS S3 KMS encryption key ID without setting it in that function too, if available. Currently in my subclass of `_object_put_params()` I overload that this way:

```
                params["ServerSideEncryption"] = self.settings.AWS_S3_ENCRYPT_KEY
                if self.settings.AWS_S3_KMS_ENCRYPTION_KEY_ID:
                    params["SSEKMSKeyId"] = self.settings.AWS_S3_KMS_ENCRYPTION_KEY_ID
```

Without that, I'd still be subclassing it. Thus the suggestion to handle keys if present.

**Comment by etianen** [comment]
Oh, I see. In that case:

AWS_S3_ENCRYPT_KEY, if a string, is the value of ServerSideEncryption. If
not a string, but it's truthy, then for backwards compatibility it is taken
to be "AES256".

AWS_S3_KMS_ENCRYPTION_KEY_ID, a new setting is the value of SSEKMSKeyId.

**Comment by etianen** [comment]
My day job has a release schedule. My open-source projects... not so much! :P

I've just released 0.12.4.

**Comment by MartinFalatic** [comment]
I totally understand 😄- thanks again!
